libraries/Microsoft.Bot.Builder.Dialogs/Memory/Scopes/DialogMemoryScope.cs

```diff
@@ -23,7 +23,7 @@ public override object GetMemory(DialogContext dc)
                 throw new ArgumentNullException(nameof(dc));
             }
 
-            return dc.Parent?.ActiveDialog?.State ?? dc.ActiveDialog.State;
+            return dc.Parent?.ActiveDialog?.State ?? dc.ActiveDialog?.State;
```

Comment:
That's a lot of question marks! I'd love if we had more constrains on the null / not null and less question marks generally